src/Backups/BackupCoordinationRemote.cpp

```diff
@@ -184,11 +184,9 @@ BackupCoordinationRemote::BackupCoordinationRemote(
             if (my_is_internal)
             {
                 String alive_node_path = my_zookeeper_path + "/stage/alive|" + my_current_host;
+                zk->createAncestors(alive_node_path);
                 auto code = zk->tryCreate(alive_node_path, "", zkutil::CreateMode::Ephemeral);
-
-                if (code == Coordination::Error::ZNODEEXISTS)
-                    zk->handleEphemeralNodeExistenceNoFailureInjection(alive_node_path, "");
-                else if (code != Coordination::Error::ZOK)
+                if ((code != Coordination::Error::ZOK) && (code != Coordination::Error::ZNODEEXISTS))
```

Comment:
I'm implementing a similar approach in https://github.com/ClickHouse/ClickHouse/pull/56917/files#diff-efd7ccd2b15733af05dd8778f80bb973e42dd8d3ad683aa8058ff3f27ed5b378R188-R195 in case you want to have a look

Reply:
For this "alive" node we don't need to check its contents, because the path to the node contains the host name.

Do you suppose it's better to always try to remove an ephemeral node from the previous session even if it's somehow exists?

Reply:
Probably yes, it's better to remove it to be sure.
So I copied this idea about removing ephemeral node from your PR, I hope you don't mind :)

Reply:
I prefer to make sure it's from this same server as otherwise you might not notice if there is something odd going on (like 2 servers with the same local hostname or some other random thing), but it's not a strong opinion.

Reply:
:+1: